tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -45,11 +45,19 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
 
     CorProfilerBase::Initialize(cor_profiler_info_unknown);
 
-    // check if tracing is completely disabled
+    // check if tracing is completely disabled.
     if (IsTracingDisabled())
     {
-        Logger::Info("DATADOG TRACER DIAGNOSTICS - Profiler disabled in ", environment::tracing_enabled);
-        return E_FAIL;
+        if (IsAzureAppServices())
+        {
+            // In AAS, the profiler is used to load other processes, so we bypass this check. If the tracer is disabled, the managed loader won't initialize instrumentation
+            Logger::Info("DATADOG TRACER DIAGNOSTICS - Ignoring that profiler is disabled as we're in AAS context");
```

Comment:
We should still stop rewriting IL, right? Except the minimum needed to to call `AgentProcessManager.Initialize()`.

Reply:
As in the loader we don't send the methods to be rewritten, we don't rewrite IL.
We still go through the callbacks that load modules but nothing more.